ui/DesignSystem/Component/Transaction/Header.svelte

```diff
@@ -0,0 +1,86 @@
+<script>
+  import {
+    formatMessage,
+    headerIcon,
+    SubjectType,
+    timestamp,
+  } from "../../../src/transaction.ts";
+
+  import { Avatar, Icon, Title, Text } from "../../../DesignSystem/Primitive";
+
+  export let avatar = null;
+  export let transaction = null;
+  export let subject = null;
+
+  const subjectAvatarShape = () => {
+    switch (subject.type) {
+      case SubjectType.User:
+      case SubjectType.Member:
+      case SubjectType.UserProject:
+        return "circle";
+      default:
+        return "square";
+    }
+  };
+
+  const icon = () => {
```

Comment:
You don't need this function at all:

```diff
diff --git a/ui/DesignSystem/Component/Transaction/Header.svelte b/ui/DesignSystem/Component/Transaction/Header.svelte
index 6e78329..60aaf3b 100644
--- a/ui/DesignSystem/Component/Transaction/Header.svelte
+++ b/ui/DesignSystem/Component/Transaction/Header.svelte
@@ -22,20 +22,6 @@
         return "square";
     }
   };
-
-  const icon = () => {
-    const val = headerIcon(transaction.messages[0]);
-    switch (val) {
-      case "Project":
-        return Icon.Source;
-      case "User":
-        return Icon.Member;
-      case "Register":
-        return Icon.Register;
-      default:
-        return Icon.Register;
-    }
-  };
 </script>

 <style>
@@ -61,7 +47,7 @@
 <div data-cy="summary" variant={transaction.id ? 'top' : 'single'}>
   <div data-cy="message" class="container">
     <div class="icon">
-      <svelte:component this={icon()} style="fill: var(--color-primary)" />
+      <svelte:component this={Icon[headerIcon(transaction.messages[0])]} style="fill: var(--color-primary)" />
     </div>
     <Title variant="big">{formatMessage(transaction.messages[0])}</Title>
     <caption style="display: inline-flex; margin:16px 0;">
diff --git a/ui/src/transaction.ts b/ui/src/transaction.ts
index 0f2d076..552dd71 100644
--- a/ui/src/transaction.ts
+++ b/ui/src/transaction.ts
@@ -277,22 +277,16 @@ export const formatMessage = (msg: Message): string => {
 export const headerIcon = (msg: Message): string => {
   switch (msg.type) {
     case MessageType.OrgRegistration:
-      return "Register";
-
     case MessageType.OrgUnregistration:
       return "Register";

-    case MessageType.MemberRegistration:
-      return "User";
-
     case MessageType.MemberUnregistration:
-      return "User";
+    case MessageType.MemberRegistration:
+    case MessageType.UserRegistration:
+      return "Member";


     case MessageType.ProjectRegistration:
-      return "Project";
-
-    case MessageType.UserRegistration:
-      return "User";
+      return "Source";
   }
 };
```

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/637/commits/a79cd89e9bc466e424ad1a72718a7a71b110a01e